src/test/java/org/akhq/repositories/TopicRepositoryTest.java

```diff
@@ -167,6 +170,20 @@ public void partition() throws ExecutionException, InterruptedException {
         assertEquals(3, topicRepository.findByName(KafkaTestCluster.CLUSTER_ID, KafkaTestCluster.TOPIC_COMPACTED).getPartitions().size());
     }
 
+    @Test
+    public void shouldThrowIfTopicDoesNotExist() {
+        final String topicName = "unknownTopic";
+        assertThrows(UnknownTopicOrPartitionException.class, () -> topicRepository.checkIfTopicExists(KafkaTestCluster.CLUSTER_ID, topicName));
```

Comment:
I'd like to verify that kafkaWrapper is indeed invoked a number of times here, but unsure as to how to inject a mocked instance of KafkaWrapperTest/AbstractKafkaWrapper into TopicRepository. 

If you have any ideas, please let me know!

Reply:
Really hard to add a test on the retry, it will be flakky since it depend on kafka server. 
Maybe we can trust the micronaut implementation and don't add a test here

Reply:
Cool, sounds okay to me - will remove